nginx/nginx.go

```diff
@@ -419,6 +421,10 @@ func createUpstreamEntries(entries controller.IngressEntries) []*upstream {
 	return sortedUpstreams
 }
 
+func validUpstreamEntry(e controller.IngressEntry) bool {
```

Comment:
Is this validation only applicable for the `nginx` updater? We could maybe have this validation done at the client or controller level otherwise we are throwing invalid ingresses around all the time. Maybe the `GetIngresses()` function in the client library might be better?

Reply:
Centralising this check makes logging the aggregated invalid ingresses easier too :smile:

Reply:
I thought about that option as well but as the issues are only related to while creating upstream config I kept it related to upstream creation only.